packages/protocol/contracts/thirdparty/automata-attestation/utils/BytesUtils.sol

```diff
@@ -0,0 +1,400 @@
+// SPDX-License-Identifier: BSD 2-Clause License
+pragma solidity 0.8.24;
+
+// Inspired by ensdomains/dnssec-oracle - BSD-2-Clause license
+// https://github.com/ensdomains/dnssec-oracle/blob/master/contracts/BytesUtils.sol
+
+library BytesUtils {
```

Comment:
Seems some inefficient code here, could be worthwhile to improve some of these and see if it actually makes a difference or not at some point (not important now).